pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -615,6 +618,39 @@ type OpenStackSDConfig struct {
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
 
+// DigitalOceanSDConfig allow retrieving scrape targets from DigitalOcean's Droplets API.
+// This service discovery uses the public IPv4 address by default, by that can be changed with relabeling
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#digitalocean_sd_config
+// +k8s:openapi-gen=true
+type DigitalOceanSDConfig struct {
+	// Authorization header configuration to authenticate against the DigitalOcean API.
+	// Cannot be set at the same time as `oauth2`.
+	// +optional
+	Authorization *v1.SafeAuthorization `json:"authorization,omitempty"`
+	// Optional OAuth 2.0 configuration.
+	// Cannot be set at the same time as `authorization`.
+	// +optional
+	OAuth2 *v1.OAuth2 `json:"oauth2,omitempty"`
+	// ProxyConfig allows customizing the proxy behaviour for this scrape config.
+	// +optional
+	*ProxyConfig `json:",inline"`
+	// Configure whether HTTP requests follow HTTP 3xx redirects.
+	// +optional
+	FollowRedirects *bool `json:"followRedirects,omitempty"`
+	// Whether to enable HTTP2.
+	// +optional
+	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
+	// TLS configuration applying to the target HTTP endpoint.
+	// +optional
+	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
+	// The port to scrape metrics from.
+	// +optional
+	Port *int `json:"port"`
```

Comment:
```suggestion
	Port *int `json:"port,omitempty"`
```

Reply:
I am not sure why it was not made `omitempty` in https://github.com/prometheus/prometheus/blob/main/discovery/digitalocean/digitalocean.go#L57 as well. In most of the discovery configs `port` field is not `omitempty`. Likely because a default value is set by Prometheus?

Reply:
Do we need to follow strictly what Prometheus has made? If we add `omitempty` we'd be following Kubernetes API conventions, as mentioned in [CONTRIBUTING.md](https://github.com/prometheus-operator/prometheus-operator/blob/main/CONTRIBUTING.md#changes-to-the-apis)

Reply:
Right, but I was curious why Prometheus left out Port field to add `omitempty` :)
I can add in here